### ISSUE TITLE
feat(api): Cap egg giving at 5 per day

### ISSUE THREAD

**Issue by cakekindel** [cakekindel/egg-party#12]
- On user gives egg, prevent giving if user has given 5 eggs since midnight
_(UTC for now, can explore using slack workspace local timezone later)_

**Note**: Blocked by #10 and/or #16 _(You need to be able to give eggs in order to cap eggs at 5 / day)_

**Comment by cakekindel** [comment]
On user gives egg, check if user has given 5 eggs since midnight

**Comment by cakekindel** [comment]
Blocked by #10 and/or #16 (Need to be able to give eggs in order to cap eggs at 5 / day)

**Comment by desertviper** [comment]
@cakekindel,

Looks like you've already started working on this:
```javascript
    /**
     * @throws {ErrorUserOutOfEggs}
     * @throws {ErrorUserTriedToGiveTooManyEggs}
     */
    public async throwIfUserCannotGiveEggs(slackUserId: string, slackWorkspaceId: string, noOfEggs: number): Promise<void>
    {
        const midnightLastNight = moment().hour(0).minute(0).second(0); // TODO: implement with moment
        const user = await this.getBySlackId(slackUserId, slackWorkspaceId);

        if (user)
        {
            const eggsGivenToday = (user.eggs || []).filter((e) => e.givenOnDate && moment(e.givenOnDate).isAfter(midnightLastNight));
            if (eggsGivenToday.length > 4) {
                throw new ErrorUserOutOfEggs();
            }

            const eggsCanGiveCount = 5 - eggsGivenToday.length;
            if (noOfEggs > 5 - eggsCanGiveCount) {
                throw new ErrorUserTriedToGiveTooManyEggs(eggsCanGiveCount, eggsGivenToday.length);
            }
        }
    }
```
I noticed you have a comment in there about refactoring it to "business layer user service".  Did you want use this story for that?

**Comment by desertviper** [comment]
Looks like you're working on #10.  Maybe I should wait and see where you land with that before making any code changes for this story.

**Comment by cakekindel** [comment]
Yeah this will be resolved by my PR, which is kind of massive and i apologize in advance

**Comment by cakekindel** [comment]
closed by #67
